experiments/nxwrapper.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""NetworkX wrapper for plotting simple tree-like Tensor Networks.
+
+A simple wrapper that creates a `networkx` graph from the nodes and edges
+of a tensor network. Works only for tensor networks that have a tree-like
+structure: the nodes are arranged to levels and only consecutive levels share
+edges. It currently has various restrictions, for example it supports dangling
+edges only at the top or the bottom level (not in the middle).
+"""
+# TODO: More general wrapper with TensorNetwork object as input.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import google_type_annotations
+from __future__ import print_function
+
+import matplotlib
+import networkx as nx
+import numpy as np
+import seaborn as sns
+from typing import List, Optional, Union, Text
+from tensornetwork import tensornetwork
+
+
+def count_dangling(node: tensornetwork.Node) -> int:
+  """Calculates the number of dangling edges of a node.
+
+  Args:
+    node: TensorNetwork node.
+
+  Returns:
+    n: Number of dangling edges this node has.
+  """
+  # TODO: Simplify this using get_all_nondangling() from #22
+  n = 0
+  for edge in node.edges:
+    if edge.is_dangling():
+      n += 1
+  return n
+
+
+class PlotWrapper(object):
+  """Wraps tree-like TensorNetwork to NetworkX for plotting."""
+  # TODO: Consider case with multiple edges between nodes.
+  # Problem with the way networkx handles multiple edges between nodes.
+  # Should also change nx.Graph() to nx.MultiGraph().
```

Comment:
Any reason you can't do this now?

Reply:
MultiGraph supports multiple edges between nodes, however they overlap in the plot and [I have not found a way to solve this](https://stackoverflow.com/questions/22785849/drawing-multiple-edges-between-two-nodes-with-networkx) (also [here](https://stackoverflow.com/questions/15053686/networkx-overlapping-edges-when-visualizing-multigraph), athough all these are a bit old).